pkg/types/json_binary_functions.go

```diff
@@ -134,6 +135,16 @@ func unquoteJSONString(s string) (string, error) {
 				}
 				char, size, err := decodeEscapedUnicode(hack.Slice(s[i+1 : i+5]))
 				if err != nil {
+					// For `surrogate pair`, it uses two `\\uxxx` to encode a character.
+					if size < 0 && len(s) >= i+10 && s[i+5] == '\\' && s[i+6] == 'u' {
```

Comment:
`size < 0` is not clear to me. I think the surrogate pair is triggered only when "invalid unicode" happens? Maybe use a specialized return value or error class to check this case.